### PR TITLE
[HttpClient] tell about ScopingHttpClient::forBaseUri()

### PR DESCRIPTION
**Pull request by nicolas-grekas**
Fixes https://github.com/symfony/symfony/issues/33274

### CODE REVIEW

File: components/http_client.rst [hunk 1]
```diff
@@ -600,25 +600,40 @@ class to autoconfigure the HTTP client based on the requested URL::
...
 
+You can define several scopes, so that each set of options is added only if a
+requested URL matches one of the regular expressions provided as keys.
+
 If the request URL is relative (because you use the ``base_uri`` option), the
 scoping HTTP client can't make a match. That's why you can define a third
 optional argument in its constructor which will be considered the default
 regular expression applied to relative URLs::
 
     // ...
```
Comment (OskarStark) [inline components/http_client.rst]: Please add the use statement for the ScopingHttpClient

File: components/http_client.rst [hunk 2]
```diff
@@ -600,25 +600,40 @@ class to autoconfigure the HTTP client based on the requested URL::
...
     );
 
+The above example can be reduced to a simpler call::
+
+    $client = ScopingHttpClient::forBaseUri($client, 'https://api.github.com/', [
+        // ...
+    ]);
+
+This way, the provided options will be used only if the requested URL is relative
+or if it matches the `https://api.github.com/` base URI.
```
Comment (OskarStark) [inline components/http_client.rst]: Please 2 backticks around

**Review by OskarStark — CHANGES_REQUESTED** [review]

File: components/http_client.rst (hunk 1 above)
Comment (nicolas-grekas) [reply components/http_client.rst]: it's already a few lines above

File: components/http_client.rst (hunk 2 above)
Comment (nicolas-grekas) [reply components/http_client.rst]: fixed

File: components/http_client.rst (hunk 1 above)
Comment (OskarStark) [reply components/http_client.rst]: Didn't see it on the phone 👍

**Review by OskarStark — APPROVED** [review]

**Comment by OskarStark** [comment]
Thank you Nicolas.

**Pull request merged**